src/API/Trace/SpanContext.php

```diff
@@ -4,92 +4,17 @@
 
 namespace OpenTelemetry\API\Trace;
 
-use OpenTelemetry\API\Trace as API;
-use function strlen;
-use function strtolower;
-
-final class SpanContext implements API\SpanContextInterface
+final class SpanContext implements SpanContextInterface
 {
     public const INVALID_TRACE = '00000000000000000000000000000000';
-    public const TRACE_VERSION_REGEX = '/^(?!ff)[\da-f]{2}$/';
     public const VALID_TRACE = '/^[0-9a-f]{32}$/';
     public const TRACE_LENGTH = 32;
     public const INVALID_SPAN = '0000000000000000';
     public const VALID_SPAN = '/^[0-9a-f]{16}$/';
-    public const SPAN_LENGTH = 16;
     public const SPAN_LENGTH_BYTES = 8;
     public const SAMPLED_FLAG = 1;
-    public const TRACE_FLAG_LENGTH = 2;
-
-    private static ?API\SpanContextInterface $invalidContext = null;
-
-    /** @inheritDoc */
-    public static function createFromRemoteParent(string $traceId, string $spanId, int $traceFlags = self::TRACE_FLAG_DEFAULT, ?API\TraceStateInterface $traceState = null): API\SpanContextInterface
```

Comment:
Given the `MUST/MUST NOT` [in the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#isremote) around remote parents, I'm in favour of keeping this method so that we're explicit about differentiating between remote/non.

Reply:
So should i stick back to my previous implementation where i created a Factory class consisting of 2 methods create and createFromRemoteParent.. Also @Nevay your thoughts on this.. before i update for one final time :D

Reply:
I'm fine with either approach.
The Java implementation provides two methods, the other implementations I checked seem to provide only one method with a `remote` parameter.

Reply:
using create and createFromRemoteParent methods defined inside SpanContextFactory class